packages/sw-lib/demo/sw.js

```diff
@@ -0,0 +1,37 @@
+/* global goog */
+
+importScripts('./sw-lib.v0.0.13.min.js');
+importScripts('./manifest.123456.js');
+
+goog.swlib.cacheRevisionedAssets(self.__file_manifest);
+
+// Specific URL defined with a string.
+goog.swlib.router.registerRoute('/example/', goog.swlib.staleWhileRevalidate());
+
+// A regular expression to catch a range of strings.
+// These only need to match part of the URL.
+goog.swlib.router.registerRoute(
+  /\/images\/(.*\/)?.*\.(png|jpg|jpeg|gif)/,
+  goog.swlib.cacheFirst()
+);
+
+// Use express style routes to capture URLs
+goog.swlib.router.registerRoute('/styles/:filename', goog.swlib.cacheFirst());
+
+// Define routes with custom cache expiration
+goog.swlib.router.registerRoute('/example-with-cache-expiration/',
+  goog.swlib.staleWhileRevalidate({
+    cacheName: 'example-cache',
+    cacheExpiration: {
+      maxEntries: 10,
+      maxAgeSeconds: 7 * 24 * 60 * 60,
+    },
+  }));
+
+// To cache third party content that doesn't support CORS, allow '0' status code
+goog.swlib.router.registerRoute('/example-third-party-caching/',
```

Comment:
Express-style routes only match same-origin requests 😨  To match third-party URLs, you're going to need to use a RegExp.

We do currently fail with an error message you if you try to pass in a full URL, like `https://cors-domain.com/path/to`, in as an Express-style route. But in this case, we would just assume that `'/example-third-party-caching/'` was a same-origin route, and it wouldn't end up matching anything.

I don't know what we could do to try to guess developer's intent here and warn them...